src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbolBase.cs

```diff
@@ -623,14 +722,35 @@ public bool HasSkipLocalsInitAttribute
         internal bool IsAutoPropertyWithGetAccessor
             => IsAutoProperty && _getMethod is object;
 
-        protected bool IsAutoProperty
-            => (_propertyFlags & Flags.IsAutoProperty) != 0;
+        internal bool IsAutoProperty
+        {
+            get
+            {
+                return (_propertyFlags & Flags.IsAutoProperty) != 0;
+            }
+        }
 
+#nullable enable
         /// <summary>
         /// Backing field for automatically implemented property, or
         /// for a property with an initializer.
         /// </summary>
-        internal SynthesizedBackingFieldSymbol BackingField { get; }
+        internal SynthesizedBackingFieldSymbol? BackingField { get; }
+
+        /// <summary>
+        /// Backing field for a semi automatically implemented property.
+        /// </summary>
+        internal SynthesizedBackingFieldSymbol? FieldKeywordBackingField
+        {
+            get
+            {
+                EnsureBackingFieldIsSynthesized();
+                var backingField = (SynthesizedBackingFieldSymbol)_lazyBackingFieldSymbolForFieldKeyword;
+                Debug.Assert(backingField is null || backingField.IsCreatedForFieldKeyword);
```

Comment:
>backingField.IsCreatedForFieldKeyword [](http://example.com/codeflow?start=53&length=37)

Instead of asserting we should be returning non null value only if ```backingField.IsCreatedForFieldKeyword``` is true.

Reply:
> Instead of asserting we should be returning non null value only if ```backingField.IsCreatedForFieldKeyword``` is true.

The implementation still doesn't do what was suggested. I think the suggestion was pretty clear.
During this PR, we encountered multiple occasions, when, I think, I gave clear instructions what to do and the instructions were not followed without any explanation or question (if instructions were not clear). What can I do to improve this situation? Is there anything I can do better?

Reply:
@AlekseyTs My bad. Sorry for that!
For this particular one I didn't notice the comment at all. Your feedback is definitely perfect and really helps a lot. Sorry again.